src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -2485,3 +2486,37 @@ class FunctionSegment(ansi.FunctionSegment):
             ),
         ),
     )
+
+
+class CreateExternalFunctionStatementSegment(BaseSegment):
+    """A `CREATE EXTERNAL FUNCTION` segment.
+
+    https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_EXTERNAL_FUNCTION.html
+    """
+
+    type = "create_external_function_statement"
+    match_grammar = Sequence(
+        "CREATE",
+        Sequence("OR", "REPLACE", optional=True),
+        "EXTERNAL",
+        "FUNCTION",
+        Ref("FunctionNameSegment"),
+        Bracketed(
+            Delimited(
+                Ref("DatatypeSegment"),
+                optional=True,
```

Comment:
I think `optional=True` is not needed? Will leave it to you though.

Reply:
It is, because there are some instances where the function has not argument. I've updated the test cases to make this clearer :+1:

Reply:
Cool, thanks!